clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -924,10 +949,19 @@ default FetchState transitionTo(FetchState newState) {
             }
         }
 
+        /**
+         * Return the valid states which this state can transition to
+         */
         Collection<FetchState> validTransitions();
 
+        /**
+         * Test if this state has a position
```

Comment:
Since the usage is a bit different, maybe we could change the name to `requiresPosition`. Then this check seems a little more intuitive:
```java
                if (this.position == null && nextState.requiresPosition()) {
                    throw new IllegalStateException("Transitioned subscription state to " + nextState + ", but position is null");
                }
```